### DISCUSSION TITLE
dtifit error & pictures not displayed in QC html report

### DISCUSSION THREAD

**Discussion by yilewang** [McIntosh-Lab/tvb-ukbb#167] [Help] [answered]
Hello everyone!

### question1:
When I ran the tvb-ukbb pipeline in the singularity container, I had some errors from `dtifit` process. The expected results should be `dti_FA.nii.gz`, `dti_V1~V3.nii.gz`. However, when I used the results generated from `dtifit` command to do 'tbss', I got the warning message `WARNING::in calculating COG, total = 0.0`, which means that `dtifit` failed. I am wondering what's a good strategy to debug this problem? I also checked the `dtifit` log, but it didn't return any error message... I was confused. Also, in the diffusion log, it didn't have any error message before `dtifit`.

```bash
2022-08-02 00:38:28,581 - bb_pipeline_diff - INFO - COMMAND TO RUN: 	${FSLDIR}/bin/dtifit -k /work/08008/yilewang/ls6/hsam/s123873/dMRI/dMRI/data_1_shell -m /work/08008/yilewang/ls6/hsam/s123873/dMRI/dMRI/nodif_brain_mask -r /work/08008/yilewang/ls6/hsam/s123873/dMRI/dMRI/data_1_shell.bvec -b /work/08008/yilewang/ls6/hsam/s123873/dMRI/dMRI/data_1_shell.bval -o /work/08008/yilewang/ls6/hsam/s123873/dMRI/dMRI/dti 
2022-08-02 00:38:29,170 - bb_pipeline_diff - INFO - COMMAND OUTPUT: 	 
2022-08-02 00:38:29,170 - bb_pipeline_diff - INFO - COMMAND TO RUN: 	$BB_BIN_DIR/bb_diffusion_pipeline/bb_tbss/bb_tbss_general s123873 
2022-08-02 00:55:01,625 - bb_pipeline_diff - INFO - COMMAND OUTPUT: 	+ direc=/work/08008/yilewang/ls6/hsam/s123873
+ TBSSDir=/work/08008/yilewang/ls6/hsam/s123873/dMRI/TBSS
+ mkdir /work/08008/yilewang/ls6/hsam/s123873/dMRI/TBSS
+ cd /work/08008/yilewang/ls6/hsam/s123873/dMRI/TBSS
+ ln ../dMRI/dti_FA.nii.gz dti_FA.nii.gz
+ /work/08008/yilewang/ls6/hsam/tvb-ukbb/bb_diffusion_pipeline/bb_tbss/bb_tbss_1_preproc dti_FA
+ /work/08008/yilewang/ls6/hsam/tvb-ukbb/bb_diffusion_pipeline/bb_tbss/bb_tbss_2_reg -T
WARNING::in calculating COG, total = 0.0
WARNING::in calculating COG, total = 0.0
WARNING::in calculating COG, total = 0.0
WARNING::in calculating COG, total = 0.0
WARNING::in calculating COG, total = 0.0
```
The image of `dti_V1.nii.gz` in fsleye:
![image](https://user-images.githubusercontent.com/37648360/182485302-83afe618-6431-454d-9793-73d02d1cefaf.png)

### question2:
And my second question is about displaying images in HTML QC report. Many of the images are missing in the QC report. However, I am pretty sure that I successfully ran all the T1 preprocessing in the structural pipeline. Could you please give me a hint about this error so that I can debug it? Thanks!!

```bash
STANDARD OUT:
HOST INFO: Linux c303-001.ls6.tacc.utexas.edu 4.18.0-305.12.1.el8_4.x86_64 #1 SMP Tue Aug 10 17:55:22 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
COMMAND RUN: /work/08008/yilewang/ls6/hsam/tvb-ukbb/tvb_bb_QC/tvb_bb_QC.sh
Starting time: Tue Aug 2 00:59:11 CDT 2022
+ set -x
+ echo ehxB
ehxB
+ set +e
++ pwd
+ origDir=/work/08008/yilewang/ls6/hsam
++ dirname /work/08008/yilewang/ls6/hsam/tvb-ukbb/tvb_bb_QC/tvb_bb_QC.sh
+ dirScript=/work/08008/yilewang/ls6/hsam/tvb-ukbb/tvb_bb_QC
++ pwd
+ dirSubject=/work/08008/yilewang/ls6/hsam/s123873
+ sub=s123873
+ [[ /work/08008/yilewang/ls6/hsam/s123873 =~ /$ ]]
+ [[ s123873 == '' ]]
+ export FSLDIR=/opt/fsl
+ FSLDIR=/opt/fsl
+ rm -f /work/08008/yilewang/ls6/hsam/s123873/QC/html/image_gen_links.js
+ echo ''

+ echo 'STARTING T1 EXTRACTION -------'
STARTING T1 EXTRACTION -------
+ /work/08008/yilewang/ls6/hsam/tvb-ukbb/tvb_bb_QC/lightbox.sh /T1/T1.nii.gz /work/08008/yilewang/ls6/hsam/s123873 T1_extraction_unmasked
unrecognized arguments: --performance 3

FSLeyes render version 1.4.6

Usage: render [options] -of outfile file [displayOpts] file [displayOpts] ...
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
OSError: [Errno 9] Bad file descriptor
unrecognized arguments: --performance 3
```

![image](https://user-images.githubusercontent.com/37648360/182484586-01ca8cfd-2913-41ce-94ef-38949bab05cc.png)

**Comment by noahfl** [comment, marked as answer]
Hey Yile! I'll do my best to answer these for you:

1) We actually have not tested and do not support TBSS for the pipeline. If you're looking for TBSS resources, [the FSL entry on it](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/TBSS/UserGuide) is a good starting point if it's something you're looking to incorporate into your workflow.

2) This is caused by a deprecated flag in `fsleyes`, `--performance 3`. In your container version (which I believe has FSL 6.0.5), the `fsleyes` version was also upgraded. Between the release in our container and the release in yours, the `fsleyes` authors [removed the `--performance` flag from `fsleyes`](https://github.com/pauldmccarthy/fsleyes/commit/9eb5f831c2233401c433370cb1f7b523501f3dfd). Removing the flag from [`tvb_bb_QC/lightbox.sh`](https://github.com/McIntosh-Lab/tvb-ukbb/blob/9b89811e3c1acde1ecc73df9ff668bbdc6533d1b/tvb_bb_QC/lightbox.sh#L200) should hopefully solve this one.

**Reply by yilewang** [reply]
ops! thanks for your reply Noah!! I apologize that I thought I had already replied to you but I was not. Following your comments, the problem is solved! Thanks for your help!

**Comment by justinliuyuwang** [comment]
Hi Yile, in case you're using the main branch, I've recently made some important changes to the pipeline for generating correct TVB inputs. Please be sure to pull the most recent version of the repository before running your data!
